vendor/bat-native-ads/src/bat/ads/internal/purchase_intent/funnel_sites.cc

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/purchase_intent/funnel_sites.h"
+
+#include "url/gurl.h"
+
+namespace ads {
+
+FunnelSites::FunnelSites() = default;
+FunnelSites::~FunnelSites() = default;
+
+FunnelSiteInfo FunnelSites::MatchFunnelSite(
+    const std::string& url) {
+  auto visited_url = GURL(url);
```

Comment:
Are we trying to match the host or just TLD+1, i.e. `help.brave.com` or just `brave.com` (TLD+1)? If we are just trying to match TLD+1 then you can use ```SameDomainOrHost(GURL(url1), GURL(url2),
      net::registry_controlled_domains::EXCLUDE_PRIVATE_REGISTRIES)``` which is part of `#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
`